### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by xdesign-diegodiniz**
This pull request introduces several new features and improvements to the Mergington High School activities application. The main changes include adding the ability for students to unregister from activities, enhancing the user interface to display and manage participants, and introducing a comprehensive test suite to ensure API reliability and correct behavior.

**Backend functionality:**

* Added an API endpoint to allow students to unregister from activities, including validation to ensure students are only removed if they are registered.
* Expanded the in-memory `activities` database to include more activities and initial participants, providing a richer dataset for both frontend and backend operations.

**Frontend improvements:**

* Updated the activities UI to display a list of participants for each activity, including an "unregister" (delete) button next to each participant.
* Implemented frontend logic to handle participant unregistration, including confirmation dialogs, API calls to the new backend endpoint, and automatic UI refresh upon changes. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R62-R105) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R127-R128)
* Enhanced CSS styles for the participants section, making the list visually appealing and improving usability for unregistering participants.

**Testing:**

* Added a comprehensive test suite for the API, covering endpoints for listing activities, signing up, unregistering, and handling edge cases such as duplicate signups and special characters in emails. [[1]](diffhunk://#diff-3451cc474d5698222168f66a293339a6bd296ee2c56f56409866bbae99917a79R1-R212) [[2]](diffhunk://#diff-001e61d97d9dee27d0c5aa1f23ba6c1972cd8a2e8320bac839656b0ab935b84dR1)

### CODE REVIEW

File: src/static/app.js [hunk 1]
```diff
@@ -20,15 +20,33 @@ document.addEventListener("DOMContentLoaded", () => {
...
         activityCard.innerHTML = `
           <h4>${name}</h4>
           <p>${details.description}</p>
           <p><strong>Schedule:</strong> ${details.schedule}</p>
           <p><strong>Availability:</strong> ${spotsLeft} spots left</p>
+          <div class="participants-section">
+            <p><strong>Participants:</strong></p>
+            ${participantsList}
+          </div>
         `;
```
[the remark this answers is not in the document]
Comment (xdesign-diegodiniz) [reply src/static/app.js]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/xdesign-diegodiniz/copilot-exam-exercise/pull/2#r2619012869)

File: tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            assert response.status_code == 200
+        
+        # Verify all students were added
+        activities_response = client.get("/activities")
+        activities_data = activities_response.json()
+        chess_participants = activities_data["Chess Club"]["participants"]
+        
+        for email in emails:
+            assert email in chess_participants
+
```
[the remark this answers is not in the document]
Comment (xdesign-diegodiniz) [reply tests/test_api.py:176]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/xdesign-diegodiniz/copilot-exam-exercise/pull/2#r2619012931)

**Comment by xdesign-diegodiniz** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/xdesign-diegodiniz/copilot-exam-exercise/pull/2#pullrequestreview-3577674147)
